src/js/pouchdb-node.js

```diff
@@ -28,19 +27,45 @@ fluid.registerNamespace("gpii.pouch.node");
 gpii.pouch.node.initDir = function (that) {
     var fullPath = path.resolve(that.options.baseDir, that.options.dbOptions.name);
 
-    // fluid.log("pouch component '" , that.id, "' saving data to '", fullPath, "'...");
+    if (!fs.existsSync(that.options.baseDir)) {
+        fs.mkdirSync(that.options.baseDir);
+        that.baseDirBelongsToUs = true;
+    }
 
-    // Right now we have to keep this synchronous so that the ordered group of "onCreate" listeners is executed correctly.
-    // TODO: Update this to use whatever we end up calling "chained" promises.
-    mkdirp.sync(fullPath);
+    if (!fs.existsSync(fullPath)) {
+        fs.mkdirSync(fullPath);
+    }
 };
 
+/**
+ *
+ * Create a "safe" prefix (i. e. db and directory name) for our content.
+ *
+ * @param toResolve An IoC reference to resolve.
+ * @returns {String} The resolved path followed by the path separator.
+ *
+ */
 gpii.pouch.node.makeSafePrefix = function (toResolve) {
     return fluid.module.resolvePath(toResolve) + path.sep;
 };
 
 /**
  *
+ * If we created our enclosing base directory, clean it up.
+ *
+ * @param that - The component itself.
+ */
+gpii.pouch.node.cleanup = function (that) {
+    if (that.baseDirBelongsToUs) {
+        fs.rmdirSync(that.options.baseDir);
```

Comment:
I'd like to see this enclosed in a no-op try/catch block to ensure that a failure doesn't interfere with any other cleanup that is on the stack - to me, this is part of the same discussion as http://stackoverflow.com/questions/130117/throwing-exceptions-out-of-a-destructor